### PR TITLE
FIX: Add Accept-Ranges header to response

### PR DESCRIPTION
**Pull request by tmacro**
Compatibility fix for AWS S3.

AWS returns a `Accept-Ranges: bytes` header with a GET or HEAD request for an object.

 ```
{
    "AcceptRanges": "bytes",
    "ContentType": "binary/octet-stream",
    "LastModified": "Tue, 10 Apr 2018 05:42:07 GMT",
    "ContentLength": 0,
    "ETag": "\"d41d8cd98f00b204e9800998ecf8427e\"",
    "Metadata": {}
}
```
Cloudserver was missing this header, which can cause programs relying upon the `Range` header to be unable to detect support.

See #1134

### CODE REVIEW

File: lib/utilities/collectResponseHeaders.js [hunk 1]
```diff
@@ -25,6 +25,8 @@ function collectResponseHeaders(objectMD, corsHeaders, versioningCfg,
     responseMetaHeaders['x-amz-version-id'] =
         getVersionIdResHeader(versioningCfg, objectMD);
 
+    responseMetaHeaders['Accept-Ranges'] = 'bytes';
```
Comment (bennettbuchanan) [inline lib/utilities/collectResponseHeaders.js]: Awesome! Thinking out loud: perhaps this could be put near the other standard HTTP headers (e.g., 'Cache-Control', etc.), just below where the Amazon-specific headers (those beginning with 'x-amz-') are set.

File: lib/utilities/collectResponseHeaders.js [hunk 2]
```diff
@@ -25,6 +25,7 @@ function collectResponseHeaders(objectMD, corsHeaders, versioningCfg,
     responseMetaHeaders['x-amz-version-id'] =
         getVersionIdResHeader(versioningCfg, objectMD);
 
+
```
Comment (JianqinWang) [inline lib/utilities/collectResponseHeaders.js]: this may be too nitpicky, but it looks like there's an extra line here

File: lib/utilities/collectResponseHeaders.js [hunk 3]
```diff
@@ -56,26 +60,26 @@ function collectResponseHeaders(objectMD, corsHeaders, versioningCfg,
...
-    Object.keys(objectMD.tags).length > 0) {
-        responseMetaHeaders['x-amz-tagging-count'] =
-          Object.keys(objectMD.tags).length;
-    }
-    if (objectMD.replicationInfo && objectMD.replicationInfo.status) {
-        responseMetaHeaders['x-amz-replication-status'] =
+        Object.keys(objectMD.tags).length > 0) {
+            responseMetaHeaders['x-amz-tagging-count'] =
+            Object.keys(objectMD.tags).length;
+        }
```
Comment (jonathan-gramain) [inline lib/utilities/collectResponseHeaders.js]: It seems the indentation is off from this point

File: lib/utilities/collectResponseHeaders.js [hunk 4]
```diff
@@ -56,26 +60,26 @@ function collectResponseHeaders(objectMD, corsHeaders, versioningCfg,
...
-        Array.isArray(objectMD.replicationInfo.backends)) {
-        objectMD.replicationInfo.backends.forEach(backend => {
-            const { status, site, dataStoreVersionId } = backend;
-            responseMetaHeaders[`x-amz-meta-${site}-replication-status`] =
+        }
+        if (objectMD.replicationInfo &&
+            // Use storageType to determine if user metadata is needed.
+            objectMD.replicationInfo.storageType &&
+            Array.isArray(objectMD.replicationInfo.backends)) {
+                objectMD.replicationInfo.backends.forEach(backend => {
```
Comment (jonathan-gramain) [inline lib/utilities/collectResponseHeaders.js]: similar indentation issue

**Comment by jonathan-gramain** [comment]
@rahulreddy As far as I know, the `Accept-Ranges` header is returned to inform of the server capabilities, it's not bound to a particular request, so it should be sent whether the object is 0-byte or not.

**Comment by jonathan-gramain** [comment]
And if some of us wonder, it's HTTP-spec related, not AWS, so it's ok even if you don't see it documented in AWS API.

**Comment by tmacro** [comment]
Here's a dump of the headers returned of a 0 byte file. `Accept-Ranges` is included

```
{  
    'accept-ranges': 'bytes',
    'content-length': '0',
    'content-type': 'binary/octet-stream',
    'date': 'Thu, 12 Apr 2018 '
            '23:34:19 GMT',
    'last-modified': 'Thu, 12 Apr '
                    '2018 23:32:09 '
                    'GMT',
    'server': 'AmazonS3',
}
```

**Comment by tmacro** [comment]
@ironman-machine
RUN_BACKBEAT_CRR_TESTS=True
try

File: lib/utilities/collectResponseHeaders.js [hunk 5]
```diff
@@ -38,12 +44,15 @@ function collectResponseHeaders(objectMD, corsHeaders, versioningCfg,
         responseMetaHeaders['x-amz-server-side-encryption-aws-kms-key-id']
             = objectMD['x-amz-server-side-encryption-aws-kms-key-id'];
     }
+
+    responseMetaHeaders['Accept-Ranges'] = 'bytes';
```
Comment (philipyoo) [inline lib/utilities/collectResponseHeaders.js:42]: I see a duplicate here

**Comment by rahulreddy** [comment]
@jonathan-gramain You would be surprised by how many times we were failing compatibility where AWS S3 was returning responses differently from the HTTP spec. It sounds common sense but not always. I usually encourage writing tests against corner cases so that it helps developers think of them.
